src/lib/components/Swap/Output.tsx

```diff
@@ -40,13 +42,22 @@ interface OutputProps {
 
 export default function Output({ disabled, children }: OutputProps) {
   const {
+    trade: { state: tradeState },
     currencyBalances: { [Field.OUTPUT]: balance },
     currencyAmounts: { [Field.INPUT]: inputCurrencyAmount, [Field.OUTPUT]: outputCurrencyAmount },
   } = useSwapInfo()
 
   const [swapOutputAmount, updateSwapOutputAmount] = useSwapAmount(Field.OUTPUT)
   const [swapOutputCurrency, updateSwapOutputCurrency] = useSwapCurrency(Field.OUTPUT)
 
+  //loading status of the trade
+  const isRouteLoading = useMemo(
+    () => TradeState.LOADING === tradeState || TradeState.SYNCING === tradeState,
+    [tradeState]
+  )
+  const independentField = useAtomValue(independentFieldAtom)
+  const showLoading = independentField === Field.INPUT && isRouteLoading
```

Comment:
See comments on naming for `Input.tsx`. Note that you should be comparing `!== Field.OUTPUT` for the output file, to stay relevant.